docs/design/ide_kpm_watcher.md

```diff
@@ -0,0 +1,78 @@
+## KCL Watch System
```

Comment:
Can you add the core API of Watch System, architecture diagram, and how to interact and integrate with KCL language server in this design document?